crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1689,7 +1689,7 @@ impl<'db> Type<'db> {
 
         if let Symbol::Type(descr_get, descr_get_boundness) = descr_get {
             let return_ty = descr_get
-                .try_call(db, &CallArguments::positional([self, instance, owner]))
+                .try_call(db, CallArgumentTypes::positional([self, instance, owner]))
```

Comment:
There are a lot of cases like this where we make a call "internally".  Here we're not really _inferring_ the types of the arguments; they're provided directly.  This is also handled nicely with the new representation: you construct `CallArgumentTypes`, since you have argument types ready to specify.